polygon/bor/finality/bor_verifier.go

```diff
@@ -84,15 +84,13 @@ func borVerify(ctx context.Context, config *config, start uint64, end uint64, ha
 	} else {
 		// in case of milestone(isCheckpoint==false) get the hash of endBlock
 		block, err := config.blockReader.BlockByNumber(ctx, roTx, end)
-		if err != nil {
+		if err != nil || block == nil {
```

Comment:
let's move it to separated PR:
- because Mark need look on it closer
- maybe I fixed it by https://github.com/ledgerwatch/erigon/pull/10173